hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -448,9 +448,9 @@ private boolean initializeFromFilesystem(String initializationTime, List<Metadat
 
       // Perform the commit using bulkCommit
       HoodieData<HoodieRecord> records = fileGroupCountAndRecordsPair.getValue();
-      bulkCommit(commitTimeForPartition, partitionType, records, fileGroupCount);
-      metadataMetaClient.reloadActiveTimeline();
       String partitionPath = (partitionType == FUNCTIONAL_INDEX || partitionType == SECONDARY_INDEX) ? dataWriteConfig.getIndexingConfig().getIndexName() : partitionType.getPartitionPath();
+      bulkCommit(commitTimeForPartition, partitionPath, records, fileGroupCount);
```

Comment:
IMO, we should remove the following and use `MetadataPartitionType#getPartitionPath` as suggested above to avoid leaking index specific logic in this metadata writer.  And we can keep passing the `MetadataPartitionType` instance as before, and use `#getPartitionPath` when needed.
```
String partitionPath = (partitionType == FUNCTIONAL_INDEX || partitionType == SECONDARY_INDEX) ? dataWriteConfig.getIndexingConfig().getIndexName() : partitionType.getPartitionPath();
```

Reply:
We can avoid most type changes in that way in the PR.

Reply:
+1

Reply:
Have done some of the refactoring. Discussed offline with Ethan - we need a separate abstraction, enum is not sufficient to pack everything. I will do that in a subsequent PR.